internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java

```diff
@@ -621,6 +621,18 @@ public JobStatusQueryResponse queryJobStatus(
     return request(ControllerRoute.JOB, params, JobStatusQueryResponse.class, timeoutMs, 1, null);
   }
 
+  /**
+   * This method will retrieve detailed job status, including uncompleted partitions and replicas from a child region.
+   * If the request is sent to a parent controller, it will be forwarded to a child controller in the specified region.
+   * This method is used for data recovery progress monitoring.
+   */
+  public JobStatusQueryResponse queryDetailedJobStatus(String kafkaTopic, String region) {
```

Comment:
This sounds good for recovery engine where both src and dst region are provided. For repush based recovery where there will be a new version for every region, can we ask the parent controller to aggregate from each region first and then return to client?

Reply:
Do we care about whether the repush completes in the other regions? For example, current version v1 has data loss in region1 but region2 and region3 are good. Then we start a repush job using data from region2. Even if the repush fails in region2 and region3, the current version v1 in region2/3 is still good (no data loss) to serve requests.

Reply:
I like this argument. Does it also means that the successfulness reported by the monitoring tool can be different from the repush result, e.g. in the above example, if region1 reports completed but region2 and region3 errored, then from the data recovery's point of view, it is completed, but from the repush job's point of view, it is not. Is my understanding correct?